ipatests/test_webui/ui_driver.py

```diff
@@ -1875,3 +1875,132 @@ def assert_action_list_action(self, action, visible=True, enabled=True,
             is_enabled = not self.has_class(li, 'disabled')
             assert is_enabled == enabled, ('Invalid enabled state of action item %s. '
                                            'Expected: %s') % (action, str(visible))
+
+    def assert_field_validation(self, expect_error, parent=None):
+        """
+        Assert for error in field validation
+        """
+
+        if not parent:
+            parent = self.get_form()
+
+        req_field_css = '.help-block[name="error_link"]'
+
+        res = self.find(req_field_css, By.CSS_SELECTOR, context=parent)
+        assert expect_error in res.text, 'No error found'
+
+    def add_multiple_record(self, entity, data1, data2, facet='search', facet_btn='add',
```

Comment:
This method is copy-paste of `add_record`. It copies some logic for data1  but not data2. It is inconsistent.

I'd suggest to rewrite it to support an unlimited number of records - accept list of data format and do all necessary actions in a loop. Thus it will have fewer code lines and be more readable.

Reply:
Fixed